### PR TITLE
Addings tests for pkg/validate.

### PR DESCRIPTION
**Pull request by kwoodson**
Adding unit tests for the `pkg/validate`.

### CODE REVIEW

File: pkg/validate/validate_test.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    secret: $AZURE_CLIENT_SECRET
+*/
+func TestValidate(t *testing.T) {
+
+	tests := map[string]struct {
+		oc            *vlabs.OpenShiftCluster
+		expectedError string
+	}{
+		"invalid version": {
+			oc: &vlabs.OpenShiftCluster{
```
Comment (pweil-) [inline pkg/validate/validate_test.go]: turn this into a `getValidOpenShiftCluster` and use the closure to make the version invalid

File: pkg/validate/validate_test.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+					},
+				},
+			},
+		},
+	}
+
+	for name, test := range tests {
+		errs := OpenShiftCluster(test.oc)
+
+		for _, err := range errs {
```
Comment (pweil-) [inline pkg/validate/validate_test.go]: check error size, it should only be 1

File: pkg/validate/validate_test.go [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+			//if err.Error != test.expectedError {
+			t.Errorf("Test[%v] failed: expected %v", name, err)
```
Comment (pweil-) [inline pkg/validate/validate_test.go]: do a contains on the string here for your error message to ensure you got the right one (will account for wrapping if we end up wrapping in a MSFT helper or something)

**Comment by jim-minter** [comment]
I e-mailed @kwoodson with a suggested alternative cleaner approach as a starting point for this

File: pkg/validate/validate_test.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+			t.Fatal(err)
+		}
+
+		if test.f != nil {
+			test.f(oc)
+		}
+
+		errs := OpenShiftCluster(oc)
+		if !reflect.DeepEqual(errs, test.expectedErrs) {
+			t.Errorf("%v: OpenShiftCluster returned unexpected result", name)
```
Comment (pweil-) [inline pkg/validate/validate_test.go]: update this to show expected and received and this is LGTM.  Nice and clean, Kenny.  I like the non-returning function better than the returning one I proposed.

Comment (pweil-) [reply pkg/validate/validate_test.go]: this can be a follow up, would like working tests merged asap.  Approving!

**Pull request merged**